tdms/tests/unit/test_interpolate.cpp

```diff
@@ -0,0 +1,128 @@
+# include "catch2/catch_test_macros.hpp"
+# include "interpolate.h"
+
+using namespace std;
+
+/**
+ * @brief Test whether checkInterpolationPoints throws exceptions in the correct cases.
+ * 
+ * This function checks whether it is possible to use cubic interpolation over a given index range.
+ */
+TEST_CASE("checkInterpolationPoints: exceptions thrown") {
+
+    // setup some fake field dimensions
+    int I = 6, J = 7, K = 8;
+
+    // 2D simulations have J = 0 - no width in this dimension
+    // checkInterpolationPoints should not throw an exception if we provide j_l = 2 > j_u = -2, as is done for a 2D simulation (J=0), and request interpolation over the maximum i,k range
+    CHECK_NOTHROW(checkInterpolationPoints(2, I - 2, 2, -2, 2, K - 2, I, 0, K));
+
+    // conversely, it should throw an error if we try to provide j_l = 0 = j_u, which one may percieve are sensible choices, in this case
+    CHECK_THROWS_AS(checkInterpolationPoints(2, I - 2, 2, 0, 0, K - 2, I, 0, K), runtime_error);
+
+    /* 3D simulation checks
+    checkInterpolationPoints should throw errors in each of the following cases:
+    - i_l < 2 with text Interpolation error: i_l too small
+    - j_l < 2 with text Interpolation error: j_l too small
+    - k_l < 2 with text Interpolation error: k_l too small
+    - i_u > I-2 with text Interpolation error: i_u too large
+    - j_u > J-2 with text Interpolation error: j_u too large
+    - k_u > K-2 with text Interpolation error: k_u too large */
+    CHECK_THROWS_AS(checkInterpolationPoints(1, I - 2, 2, J - 2, 2, K - 2, I, J, K), runtime_error);
+    CHECK_THROWS_AS(checkInterpolationPoints(2, I - 1, 2, J - 2, 2, K - 2, I, J, K), runtime_error);
+    CHECK_THROWS_AS(checkInterpolationPoints(2, I - 2, 0, J - 2, 2, K - 2, I, J, K), runtime_error);
+    CHECK_THROWS_AS(checkInterpolationPoints(2, I - 2, 2, J - 0, 2, K - 2, I, J, K), runtime_error);
+    CHECK_THROWS_AS(checkInterpolationPoints(2, I - 1, 2, J - 2, -1, K - 2, I, J, K), runtime_error);
+    CHECK_THROWS_AS(checkInterpolationPoints(2, I - 1, 2, J - 2, 2, K + 1, I, J, K), runtime_error);
+
+    /* When provided input arguments that do not satisfy any of the above - even cases such as i_l > i_u - should pass without throwing an exception */
```

Comment:
could maybe separate these out into another test with a more informative name

Reply:
Have taken the "no exceptions thrown" checks out into their own test_case